### PR TITLE
Fix/updating linters

### PR DESCRIPTION
**Pull request by GabrielAzevedoFerreiraQB**
## Motivation and Context
Why was this PR created?
- [updating Black version](https://github.com/quantumblacklabs/causalnex/pull/156/commits/96c4fddb1c38dc743ed431cd6213f9192a51f0ef)
    - I was seeing this error before: 
image
    
- [linting - adapting to pylint updates to newest version](https://github.com/quantumblacklabs/causalnex/pull/156/commits/f5c0e0ce8dee7cd73b778ea53ffcdc169d249c08)
There were these linting issues  
image
- Docs wasn't building: I added as requirements:

````
ipython_genutils
Markupsafe<2.1
````

## How has this been tested?
What testing strategies have you used?

## Checklist

- [ ] Read the [contributing](https://github.com/quantumblacklabs/causalnex/blob/develop/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change and added my name to the list of supporting contributions in the [`RELEASE.md`](https://github.com/quantumblacklabs/causalnex/blob/develop/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Assigned myself to the PR

## Notice

- [ ] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -41,6 +41,7 @@ utils:
   setup_pre_commit: &setup_pre_commit
     name: Install pre-commit hooks
     command: |
+      pip install --upgrade pre-commit
```
Comment (GabrielAzevedoFerreiraQB) [inline .circleci/config.yml:44]: with the new version of black, if we don't add this line, we see this:

image

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -29,7 +29,7 @@ repos:
         exclude: ^causalnex/ebaybbn
 
 -   repo: https://github.com/psf/black
-    rev: 20.8b1
+    rev: 22.3.0
```
Comment (GabrielAzevedoFerreiraQB) [inline .pre-commit-config.yaml:32]: If not upgraded we see the error in the PR description.

But, `22.3.0` linting rules may change slightly. in our case it changed only `a ** b` to `a**b` in a few places

File: .circleci/config.yml (hunk 1 above)
Comment (oentaryorj) [reply .circleci/config.yml:44]: I guess this will always upgrade to the latest `pre-commit` version? Is it better to specify the version elsewhere?

Comment (GabrielAzevedoFerreiraQB) [reply .circleci/config.yml:44]: Good point
Actually, I have to update this on the test requirements instead. Thanks for the push


image

**Pull request merged**